packages/flutter/lib/src/widgets/icon_theme_data.dart

```diff
@@ -106,7 +114,8 @@ class IconThemeData with Diagnosticable {
     return other is IconThemeData
         && other.color == color
         && other.opacity == opacity
-        && other.size == size;
+        && other.size == size
+        && const ListEquality<Shadow>().equals(other.shadows, shadows);
```

Comment:
please just use `listEquals` from 'package:flutter/foundation.dart' and remove the import of collections above.